app/repositories/OptOutSessionDataRepository.scala

```diff
@@ -94,13 +94,29 @@ class OptOutSessionDataRepository @Inject()(val repository: UIJourneySessionData
     }
   }
 
-  def initialiseOptOutJourney(oop: OptOutProposition)(implicit hc: HeaderCarrier): Future[Boolean] = {
+  def initialiseOptOutJourney(oop: OptOutProposition, shouldResetIntent: Boolean = true)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[Boolean] = {
     val data = UIJourneySessionData(
       sessionId = hc.sessionId.get.value,
       journeyType = OptOutJourney.toString,
       optOutSessionData = Some(OptOutSessionData(Some(buildOptOutContextData(oop)), selectedOptOutYear = None))
     )
-    repository.set(data)
+    if(shouldResetIntent){
+      val dataWithResetIntent = data.copy(
+        optOutSessionData = data.optOutSessionData.map(_.copy(selectedOptOutYear = None))
+      )
+      repository.set(dataWithResetIntent)
+    }else {
```

Comment:
It's a tiny tiny nitpick but I think some of the spacing between brackets is incorrect